I have implemented a new test and verified the regression in the master branch. I ran this new test (without the call to DeleteRemovedPods) against the current `master` branch, and it fails with the original issue's signature.

Here is the terminal output from the failure on `master` for reference:

```
I0704 08:17:40.871693  146051 cluster_feeder.go:546] "Failed to record OOM" oomInfo={"Timestamp":"2026-07-04T08:18:40.871210937+05:30","Memory":2147483648,"ContainerID":{"Namespace":"default","PodName":"Pod2","ContainerName":"containerB"}} error="KeyError: {default Pod2}"
--- FAIL: TestClusterStateFeeder_LoadRealTimeMetrics_OOMEventsWithDeletedPods (0.00s)
    /home/naveen/Documents/autoscaler/vertical-pod-autoscaler/pkg/recommender/input/cluster_feeder_test.go:814: 
        	Error Trace:	/home/naveen/Documents/autoscaler/vertical-pod-autoscaler/pkg/recommender/input/cluster_feeder_test.go:814
        	Error:      	"map[{default Pod1}:0x2d20d8944060]" should have 2 item(s), but has 1
        	Test:       	TestClusterStateFeeder_LoadRealTimeMetrics_OOMEventsWithDeletedPods
    /home/naveen/Documents/autoscaler/vertical-pod-autoscaler/pkg/recommender/input/cluster_feeder_test.go:815: 
        	Error Trace:	/home/naveen/Documents/autoscaler/vertical-pod-autoscaler/pkg/recommender/input/cluster_feeder_test.go:815
        	Error:      	map[model.PodID]*model.PodState{model.PodID{Namespace:"default", PodName:"Pod1"}:(*model.PodState)(0x2d20d8944060)} does not contain model.PodID{Namespace:"default", PodName:"Pod2"}
        	Test:       	TestClusterStateFeeder_LoadRealTimeMetrics_OOMEventsWithDeletedPods
    /home/naveen/Documents/autoscaler/vertical-pod-autoscaler/pkg/recommender/input/cluster_feeder_test.go:823: 
        	Error Trace:	/home/naveen/Documents/autoscaler/vertical-pod-autoscaler/pkg/recommender/input/cluster_feeder_test.go:823
        	Error:      	"1288490188" is not greater than "1288490188"
        	Test:       	TestClusterStateFeeder_LoadRealTimeMetrics_OOMEventsWithDeletedPods
FAIL
FAIL	k8s.io/autoscaler/vertical-pod-autoscaler/pkg/recommender/input	0.011s
FAIL```

--- a/vertical-pod-autoscaler/pkg/recommender/input/cluster_feeder.go
+++ b/vertical-pod-autoscaler/pkg/recommender/input/cluster_feeder.go
@@ -96,6 +96,7 @@ type ClusterStateFeederFactory struct {
 	RecommenderName     string
 	IgnoredNamespaces   []string
 	VpaObjectNamespace  string
+	podsToDelete        []model.PodID
 }
 
 // Make creates new ClusterStateFeeder with internal data providers, based on kube client.
@@ -115,6 +116,7 @@ func (m ClusterStateFeederFactory) Make() *clusterStateFeeder {
 		recommenderName:     m.RecommenderName,
 		ignoredNamespaces:   m.IgnoredNamespaces,
 		vpaObjectNamespace:  m.VpaObjectNamespace,
+		podsToDelete:        m.podsToDelete,
 	}
 }
 

